### PR TITLE
Make sure into_raw_fd doesn't close the file descriptor

### LINKED ISSUE #147: socket closed after UnixStream::into_raw_fd()
UnixStream holds a `IoHandle<mio_uds::UnixStream>` which is not correctly disposed of in `into_raw_fd`, resulting in the returned file descriptor being closed when `into_raw_fd` returns, making the function useless.

### PR DESCRIPTION
**Pull request by yshui**
Closes #147

### CODE REVIEW

File: src/net/driver/mod.rs [hunk 1]
```diff
@@ -278,13 +278,26 @@ impl<T: Evented> IoHandle<T> {
 
         Ok(())
     }
+
+    /// Deregister and return the I/O source
```
Comment (stjepang) [inline src/net/driver/mod.rs]:
```suggestion
    /// Deregisters and returns the inner I/O source.
```

File: src/net/driver/mod.rs [hunk 2]
```diff
@@ -278,13 +278,26 @@ impl<T: Evented> IoHandle<T> {
...
+    ///
+    /// This method is to support IntoRawFd in struct that uses IoHandle
```
Comment (stjepang) [inline src/net/driver/mod.rs]:
```suggestion
    /// This method is typically used to convert `IoHandle`s to raw file descriptors/handles.
```

**Comment by yshui** [comment]
@stjepang Thanks. Applied

**Pull request merged**